proposed/2023/NuGetAudit-without-nuget.org-as-packageSource.md

```diff
@@ -0,0 +1,221 @@
+It would be worse because for customers who do not have nuget.org blocked at the firewall, it would require a manual action to update the VDB, whereas an audit source would allow NuGet to automatically download and cache the VDB.
+
+### Specify audit sources via MSBuild
+
+Some customers are MSBuild enthusiasts and wish to be able to control everything via MSBuild.
+In theory, this could allow customers to avoid having a *nuget.config* file in their repo.
+However, there are multiple features in NuGet that only work via *nuget.config*, such as package source credentials, and package source mapping.
+Therefore, in order to avoid implementation delays, we'll limit the scope 
+
+Additionally, the biggest benefit to MSBuild is that different projects can use different values.
+However, it's not clear what customer benefit would result in being able to use different audit sources in different projects in the same solution or repo.
+My best guess is that the most likely scenario is customers using different settings when working in the office compared to working from home.
+However, we have not received much feedback that this is desirable for package sources, so it's unlikely that it's important for audit sources.
+
+## Prior Art
+
+<!-- What prior art, both good and bad are related to this proposal? -->
+<!-- Do other features exist in other ecosystems and what experience have their community had? -->
+<!-- What lessons from other communities can we learn from? -->
+<!-- Are there any resources that are relevant to this proposal? -->
+Although [`npm audit`'s help docs](https://docs.npmjs.com/cli/v10/commands/npm-audit?v=true) don't mention it, it appears to have a `--registry={url}` argument, as can be seen on [the Azure DevOps docs on npm audit](https://learn.microsoft.com/azure/devops/artifacts/npm/npm-audit?view=azure-devops&tabs=yaml).
+
+Pypi's `pip` has [an audit command](https://pypi.org/project/pip-audit/), and `--index-url` and `--extra-index-url` appear to enable using custom audit sources.
+
+Rust's cargo audit command has [configuration for the advisory DB URL](https://docs.rs/cargo-audit/latest/cargo_audit/config/struct.DatabaseConfig.html)
+
+Note that for these other package ecosystems, they do not appear to need any explicit configuration file in order to specify default sources (either package source, or audit source).
+Instead, defaults appear hard-coded into the app, and it requires explicit configuration to remove.
+
+Contrast this to NuGet's design where NuGet creates a default configuration file at first use, and then requires sources to be explicitly defined.
+However, NuGet settings that fall under the `<config>` section of the *nuget.config* file do have hard-coded defaults that do not need to be explicitly defined in any config file.
+
+Both choices have advantages and disadvantages.
```

Comment:
That's a great discussion point, @zivkan . When a new nuget.config is created, should it be created with nuget.org as the default audit source? IMHO, it should, so that in case a user removes nuget.org from the packageSources list, they will continue getting the auditing functionality.